src/main/java/uk/gov/hmcts/reform/adoption/adoptioncase/caseworker/event/page/SendOrReply.java

```diff
@@ -0,0 +1,146 @@
+package uk.gov.hmcts.reform.adoption.adoptioncase.caseworker.event.page;
+
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.collections4.CollectionUtils;
+import uk.gov.hmcts.ccd.sdk.api.CaseDetails;
+import uk.gov.hmcts.ccd.sdk.api.callback.AboutToStartOrSubmitResponse;
+import uk.gov.hmcts.ccd.sdk.type.DynamicList;
+import uk.gov.hmcts.ccd.sdk.type.DynamicListElement;
+import uk.gov.hmcts.reform.adoption.adoptioncase.model.CaseData;
+import uk.gov.hmcts.reform.adoption.adoptioncase.model.State;
+import uk.gov.hmcts.reform.adoption.adoptioncase.model.SelectedMessage;
+import uk.gov.hmcts.reform.adoption.adoptioncase.model.MessageDocumentList;
+import uk.gov.hmcts.reform.adoption.adoptioncase.model.MessageSendDetails;
+import uk.gov.hmcts.reform.adoption.common.ccd.CcdPageConfiguration;
+import uk.gov.hmcts.reform.adoption.common.ccd.PageBuilder;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+
+@Slf4j
+public class SendOrReply implements CcdPageConfiguration {
+    @Override
+    public void addTo(PageBuilder pageBuilder) {
+        pageBuilder
+            .page("pageSendOrReply1", this::midEvent)
+            .mandatory(CaseData::getMessageAction)
+            .mandatory(CaseData::getReplyMsgDynamicList, "messageAction=\"replyMessage\"")
+            .page("pageSendOrReply3")
+            .showCondition("messageAction=\"replyMessage\"")
+            .label("labelReplyMes", "## Reply a message")
```

Comment:
Should this label be 'Reply to message'? pls check figma for this

Reply:
Done